Sources/WebSocketKit/Concurrency/WebSocket+Concurrency.swift

```diff
@@ -60,19 +64,41 @@ extension WebSocket {
         }
     }
 
-    @preconcurrency public func onPong(_ callback: @Sendable @escaping (WebSocket) async -> ()) {
+    @preconcurrency public func onPong(_ callback: @Sendable @escaping (WebSocket, ByteBuffer) async -> ()) {
+        self.eventLoop.execute {
+            self.onPong { socket, data in
+                Task {
+                    await callback(socket, data)
+                }
+            }
+        }
+    }
+    
+    @available(*, deprecated, message: "Please use `onPong { socket, data in /* … */ }` with the additional `data` parameter.")
+    public func onPong(_ callback: @Sendable @escaping (WebSocket) async -> ()) {
```

Comment:
This needs annotating with `@preconcurrency` to avoid breaking anyway using unsafe code in the callbak
```suggestion
    @preconcurrency public func onPong(_ callback: @Sendable @escaping (WebSocket) async -> ()) {
```